src/storage/async_storage/adapter.rs

```diff
@@ -0,0 +1,145 @@
+// src/storage/adapter.rs
```

Comment:
I actually like this filename a bit better -- I don't think the async functionality needs to be isolated in a submodule.